pkg/controller/certificates/csrapprover.go

```diff
@@ -221,16 +241,79 @@ func isYurtCSR(csr *certificates.CertificateSigningRequest) bool {
 
 // checkCertApprovalCondition checks if the given csr's status is
 // approved or denied
-func checkCertApprovalCondition(
-	status *certificates.CertificateSigningRequestStatus) (
-	approved bool, denied bool) {
+func checkCertApprovalCondition(status *certificatesv1.CertificateSigningRequestStatus) (approved bool, denied bool) {
 	for _, c := range status.Conditions {
-		if c.Type == certificates.CertificateApproved {
+		if c.Type == certificatesv1.CertificateApproved {
 			approved = true
 		}
-		if c.Type == certificates.CertificateDenied {
+		if c.Type == certificatesv1.CertificateDenied {
 			denied = true
 		}
 	}
 	return
 }
+
+func updateApproval(ctx context.Context, client kubernetes.Interface, csr *certificatesv1.CertificateSigningRequest) error {
+	_, v1err := client.CertificatesV1().CertificateSigningRequests().UpdateApproval(ctx, csr.Name, csr, metav1.UpdateOptions{})
+	if v1err == nil || !apierrors.IsNotFound(v1err) {
+		return v1err
+	}
+
+	v1beta1Csr := v1Csr2v1beta1Csr(csr)
+	_, v1beta1err := client.CertificatesV1beta1().CertificateSigningRequests().UpdateApproval(ctx, v1beta1Csr, metav1.UpdateOptions{})
+	return v1beta1err
+}
+
+func v1Csr2v1beta1Csr(csr *certificatesv1.CertificateSigningRequest) *certificatesv1beta1.CertificateSigningRequest {
```

Comment:
How about use conversion func like `Convert_v1_CertificateSigningRequestList_To_certificates_CertificateSigningRequestList` which provided by k8s?

Reply:
but we need to import k8s.io/kubernetes, so we provide these funcs.